kernel/src/process/signal/mod.rs

```diff
@@ -201,10 +207,24 @@ pub fn handle_user_signal(
 
     // 2. Write ucontext_t.
 
+    // Save the current signal stack information.
+    let uc_stack = {
+        let mut sig_stack = ctx.thread_local.sig_stack().borrow_mut();
+        let stack = stack_t::from(&*sig_stack);
```

Comment:
How does the code here interact with `SIG_STACK_FLAGS_MASK` (which is used in `get_old_stack`)?

Reply:
The stack information in uc_context directly reflects the user-set flags, so it behaves differently from get_old_stack and does not use the SIG_STACK_FLAGS_MASK.

I have added a new test to verify this behavior. The test invokes a signal handler that does not execute on the signal stack, demonstrating that the stack flags in uc_context match those set by the user. It also shows that the SS_ONSTACK flag can be set in this context even if the stack is inactive.

Reply:
Thanks for the explanation and the experiments!